src/MoltenFunding.sol

```diff
@@ -89,22 +99,30 @@ contract MoltenFunding is ReentrancyGuard {
         require(totalDeposited > 0, "Molten: no deposits");
 
         exchangeTime = block.timestamp;
-        exchangeRate = _exchangeRate;
+        exchangeRate = queryExchangeRate(totalDeposited);
+        assert(exchangeRate < type(uint128).max);
 
-        mToken.mint(address(this), daoTokenTotal);
+        mToken.mint(
+            address(this),
+            (totalDeposited * 10**mToken.decimals()) / exchangeRate
```

Comment:
Maybe set this value in a variable to avoid spending gas on this calculation again below.

Reply:
I don't see any way to save on gas. Please note that this line is looking at `mToken` decimals, whereas the other one looks at `daoToken`. Also, I could calculate `totalDeposited / exchangeRate` beforehand but as fractional numbers are not supported out of the box, I'm not sure this is worth it (in terms of both effort and gas).

Reply:
Missed the distinction between `daoTokens` and `mTokens` 😅. Disregard.